Description: Fix FTBFS on arm64, mips/mips64/mipsel, ppc64el, s390x
Origin: upstream, https://github.com/supertuxkart/stk-code/commit/5e05f1178ce6bc5f3a653b55ab3dc6d016196341
Bug: https://github.com/supertuxkart/stk-code/issues/2569
Bug-Debian: https://bugs.debian.org/830748
Last-Update: 2016-12-03

--- a/lib/angelscript/projects/cmake/CMakeLists.txt
+++ b/lib/angelscript/projects/cmake/CMakeLists.txt
@@ -68,6 +68,7 @@ set(ANGELSCRIPT_SOURCE
     ../../source/as_bytecode.cpp
     ../../source/as_callfunc.cpp
     ../../source/as_callfunc_x86.cpp
+    ../../source/as_callfunc_mips.cpp
     ../../source/as_callfunc_x64_gcc.cpp
     ../../source/as_callfunc_x64_msvc.cpp
     ../../source/as_callfunc_x64_mingw.cpp
--- a/lib/angelscript/source/as_config.h
+++ b/lib/angelscript/source/as_config.h
@@ -844,7 +844,7 @@
 			#define THISCALL_PASS_OBJECT_POINTER_ON_THE_STACK
 			#define AS_X86
 			#undef AS_NO_THISCALL_FUNCTOR_METHOD
-		#elif defined(__LP64__) && !defined(__arm64__)
+		#elif defined(__x86_64__)
 			#define AS_X64_GCC
 			#undef AS_NO_THISCALL_FUNCTOR_METHOD
 			#define HAS_128_BIT_PRIMITIVES
